docs/week-3.md

```diff
@@ -0,0 +1,523 @@
+`useState` for our message example:
+
+```
+const [profileLikes, setProfileLikes] = useState<Number>(0);
+const [message, setMessage] = useState<String>("Hello World")
+```
+
+When typing components, we have to make sure to type the props that we are taking in (if applicable). The way
+we do this is with an interface with the name and the type of the prop we are expecting. For example lets 
+take the `LuckyNum` component as an example.
+
+```
+import React from 'react'; 
+
+/*
+    We specify luckyNumber as readonly because we do not need to modify
+    it anywhere in our component. All we need is to be able to read the value.
+*/
+
+interface LuckyNumProps {
+    readonly luckyNumber: number;
+}
+
+/*
+    React.FC = react functional component
+*/
+
+const LuckyNum: React.FC<LuckyNumProps> = (props) => (
+
+    <div>
+        <h1>My Lucky Number is {props.luckyNumber}</h1>
+    </div>
+
+)
+
+export default LuckyNum;
+```
+
+## Material UI: a lifesaver
+We all hate having to style our components for hours on end to make them look beautiful and actually work
+properly. Do not worry because Material UI is here to save the day!
+
+### What is Material UI?
+Material UI is a popular framework in react. It is a library filled with pre-styled components that are super
+helpful. There are many different components that basically replace the need for any pure html components in
+your project. Each of the pre-made components will save you a ton of time on styling and formatting.
+
+### How can I use it?
+All you have to do is import any component you want from the list found 
+[here](https://material-ui.com/components/box/). For example let us replace the html `<button>` for John's 
+likes with a Material UI Button component.
+
+```
+import React, { useState } from 'react';
+import LuckyNum from './LuckyNum';
+import { Button } from '@material-ui/core'
+
+const Profile = () => (
+
+    //initialize the state to 0 because we start with 0 likes
+    const [profileLikes, setProfileLikes] = useState(0);
+
+    //handles incrementing the likes of John's profile
+    function handleLikes {
+        setProfileLikes(profileLikes + 1);
+    }    
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <Button onClick={()=>handleLikes}>Likes: {profileLikes}</Button>
+    </div>
+);
+
+export default Profile;
+```
+
+### Styling Material UI Components
+If you wanted to style the components in Material UI a little more to taylor your specific needs you can do that!
+For example let us make the newly imported Button component on John's profile to be black.
+
+We first have to import `makeStyles` from `@material-ui/core/styles`. We then make an object for each class we 
+want to apply to our components. We make a lambda function called `useStyles` that is equal to `makeStyles`
```

Comment:
what is a lambda function?

Reply:
should have been covered in Week 2

Reply:
However, a lambda function is a variable in JS that acts like an anonymous function.

Reply:
We called them anonymous functions in week 2, use the same terminology here